R/while_oai.R

```diff
@@ -16,26 +16,41 @@ while_oai <- function(url, args, token, as, dumper=NULL, dumper_args=NULL, ...)
     res <- GET(url, query = args2, ...)
     stop_for_status(res)
     tt <- content(res, "text")
-    xml_orig <- xml2::read_xml(tt)
-    handle_errors(xml_orig)
-    xml <- xml2::xml_children(xml2::xml_children(xml_orig)[[3]])
-    trytok <- xml2::as_list(xml[sapply(xml, xml_name) == "resumptionToken"])
-    if (length(trytok) == 0) {
-      tok <- list(token = "")
+
+    # try parsing
+    parsed <- try( read_xml_safely(tt), silent=TRUE )
```

Comment:
do you intend for user to be able to pass  `xml_invalid_as` and any options through `...`?